Using `PydanticAIPlugin` triggers mypy error `no-untyped-call`

### Initial Checks

- [x] I'm using the [latest version](https://github.com/pydantic/pydantic-ai/releases/latest) of Pydantic AI
- [x] I've searched for my issue in [the issue tracker](https://github.com/pydantic/pydantic-ai/issues) before opening this issue

### Description

Using `PydanticAIPlugin` and checking with mypy under strict mode results in a `no-untyped-call` error.

```console
$ uvx --with "mypy" --with "pydantic-ai[temporal]" mypy --strict mre.py
mre.py:3: error: Call to untyped function "PydanticAIPlugin" in typed context  [no-untyped-call]
Found 1 error in 1 file (checked 1 source file)
```

The expected behaviour is that there is no mypy error (similar to how there isn't an error when using `LogfirePlugin`, for example).

This is reproducible when using the `pydantic.mypy` plugin as well.

### Minimal, Reproducible Example

```Python
from pydantic_ai.durable_exec.temporal import PydanticAIPlugin

PydanticAIPlugin()
```

### Logfire Trace

_No response_

### Python, Pydantic AI & LLM client version

- Python: 3.13.0
- Pydantic AI: 2.0.0
- LLM provider SDK: N/A
- mypy: 2.1.0

--- a/pydantic_ai_slim/pydantic_ai/durable_exec/temporal/__init__.py
+++ b/pydantic_ai_slim/pydantic_ai/durable_exec/temporal/__init__.py
@@ -116,7 +116,7 @@ def _workflow_runner(runner: WorkflowRunner | None) -> WorkflowRunner:
 class PydanticAIPlugin(SimplePlugin):
     """Temporal client and worker plugin for Pydantic AI."""
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__(  # type: ignore[reportUnknownMemberType]
             name='PydanticAIPlugin',
             data_converter=_data_converter,

